lib/repos/game.rb

```diff
@@ -31,66 +31,31 @@ def pretty_print
       end
       # user: TicTac::Repos::User
       def self.create(id,opponent,game_class)
-        @game=game_class.new_game
+        @game=game_class.new
         chain=TicTac::Block.from_data(id,nil,{game: game_class.name,
                                               players:[opponent,id.public_key_link],
-                                              game_state: @game.marshal,
+                                              state: @game.state
                                              }).ipfs_addr
         return Game.new(chain)
       end
       def get_player_index(ipfs_link)
         idx=@players.index(ipfs_link)
         if !idx
-          GameError("INVALID_PLAYER")
+          raise GameError.new("INVALID_PLAYER")
         end
         idx
       end
       def process_move(player_key,move)
         player=get_player_index(player_key)
-        @game.move(player,move)
+        @game=@game.move(player,move)
       end
       def move(id,move)
         process_move(id.public_key_link,move)
-        Game.new(@chain.last.append(id,move).ipfs_addr)
+        @game=Game.new(@chain.last.append(id,move).ipfs_addr)
```

Comment:
mm this isn't ideal. The `Game` model should be able to play a game in the abstract, and this repo should be keeping track of the `Game` and adding to it as we continue to receive moves. Also it probably shouldn't use game specific idioms like `move` but instead use appendlog specific idioms like `append`